app/controllers/api/v1/games/ships_controller.rb

```diff
@@ -3,29 +3,13 @@ module V1
     module Games
       class ShipsController < ApiController
         def create
-          game = Game.find(params[:game_id])
           ship = Ship.new(ship_params[:ship_size])
-
-          ship_placer = if request.headers['X-API-Key'] == game.player_1.api_key
-            binding.pry
-                          ShipPlacer.new(
-                            board: game.player_1.board,
-                            ship: ship,
-                            start_space: ship_params[:start_space],
-                            end_space: ship_params[:end_space]
-                          ).run
-                          game.save
-                        else
-                          ShipPlacer.new(
-                            board: game.player_2.board,
-                            ship: ship,
-                            start_space: ship_params[:start_space],
-                            end_space: ship_params[:end_space]
-                          ).run
-                          game.save
-                        end
-          # binding.pry
-          render json: game, message: ship_placer
+          ship_placer = ShipPlacer.new({ board: current_player.board,
+                                         ship: ship,
+                                         start_space: ship_params[:start_space],
+                                         end_space: ship_params[:end_space] }).run
+          current_game.save
+          render json: current_game, message: ship_placer
         end
```

Comment:
So since I set the current_player in the API controller we are able to just set their board as the board entering the ship_placer initialize arguement. Everything else here pretty much stayed the same. But this also allowed me to extract the game.save method (which became current_game.save) because it was giving us a return value of either true or false, which was interfering with out ship_placer message.